core/src/main/java/org/polypheny/db/catalog/entity/CatalogDataPlacement.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2019-2022 The Polypheny Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.polypheny.db.catalog.entity;
+
+
+import com.google.common.collect.ImmutableList;
+import java.io.Serializable;
+import java.sql.Timestamp;
+import lombok.NonNull;
+import lombok.Setter;
+import lombok.SneakyThrows;
+import org.polypheny.db.catalog.Catalog;
+import org.polypheny.db.catalog.Catalog.DataPlacementRole;
+import org.polypheny.db.catalog.Catalog.PlacementType;
+
+
+/**
+ * Serves as a container, which holds all information related to a table entity placed on physical store.
+ */
+public class CatalogDataPlacement implements CatalogEntity {
+
+    private static final long serialVersionUID = 3758054726464326557L;
+    public final long tableId;
+    public final int adapterId;
+
+    public final PlacementType placementType;
+
+    // Is present at the DataPlacement && the PartitionPlacement
+    // Although, partitionPlacements are those that get effectively updated
+    // A DataPlacement can directly forbid that any Placements within this DataPlacement container can get outdated.
+    // Therefore, the role at the DataPlacement specifies if underlying placements can even be outdated.
+    public final DataPlacementRole dataPlacementRole;
+
+    public final ImmutableList<Long> columnPlacementsOnAdapter;
+    public final ImmutableList<Long> partitionPlacementsOnAdapter;
+
+
+    // The newest commit timestamp when any partitions inside this placement has been updated or refreshed
+    // Equals the newest timestamp ony any of the CatalogPartitionPlacements.
+    // Technically other  linked attachments could still have older update timestamps.
+    // This should help to quickly identify placements that can fulfil certain conditions.
+    // Without having to traverse all partition placements one-by-one
+    @Setter
+    public Timestamp updateTimestamp;
```

Comment:
All catalog objects should be immutable. 

See: https://mapdb.org/book/quick-start/

> Data stored in MapDB (keys and values) should be immutable. MapDB serializes objects on background.

Reply:
`updateTimestamp` has been removed